modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx

```diff
@@ -970,6 +971,139 @@ const MyPopup = () => {
 }
 ```
 
+### Modal
+
+Modal has transitioned to a [compound component](./COMPOUND_COMPONENTS.mdx). What was `Modal` in v4
+is now `Modal.Card` in v5.
+
+#### v4
+
+```tsx
+import React from 'react';
+
+import {Modal} from '@workday/canvas-kit-react-modal';
+import {DeleteButton, Button} from '@workday/canvas-kit-react-button';
+
+const MyModal = () => {
+  const handleDelete = () => {
+    console.log('Deleted item');
+  };
+
+  const {targetProps, modalProps, closeModal} = useModal();
+
+  return (
+    <>
+      <DeleteButton {...targetProps}>Delete Item</DeleteButton>
+      <Modal heading={'Delete Item'} {...modalProps}>
+        <p>Are you sure you want to delete the item?</p>
+        <DeleteButton
+          style={{marginRight: '16px'}}
+          onClick={() => {
+            closeModal();
+            handleDelete();
+          }}
+        >
+          Delete
+        </DeleteButton>
+        <Button onClick={closeModal} variant={Button.Variant.Secondary}>
+          Cancel
+        </Button>
+      </Modal>
+    </>
+  );
+};
+```
+
+#### v5
+
+```tsx
+import React from 'react';
+
+import {Modal} from '@workday/canvas-kit-react/modal';
+import {DeleteButton} from '@workday/canvas-kit-react/button';
+import {HStack} from '@workday/canvas-kit-labs-react';
+
+const MyModal = () => {
+  const handleDelete = () => {
+    console.log('Deleted item');
+  };
+
+  return (
+    <Modal>
+      <Modal.Target as={DeleteButton}>Delete Item</Modal.Target>
+      <Modal.Overlay>
+        <Modal.Card>
+          <Modal.CloseIcon aria-label="Close" />
+          <Modal.Heading>Delete Item</Modal.Heading>
+          <Modal.Body>
+            <p>Are you sure you want to delete the item?</p>
+            <HStack spacing="s">
+              <Modal.CloseButton as={DeleteButton} onClick={handleDelete}>
+                Delete
+              </Modal.CloseButton>
+              <Modal.CloseButton>Cancel</Modal.CloseButton>
+            </HStack>
+          </Modal.Body>
+        </Modal.Card>
+      </Modal.Overlay>
+    </Modal>
+  );
+};
+```
+
+Most notably, `Modal` is now a container component that takes a `ModalModel` and has several
+subcomponents. `Modal` looks much like the structure of [Popups](#popups), except `Modal` has a
+`Modal.Overlay` subcomponent instead of a `Popup.Popper` component. The `Modal.Overlay` is the
+component in charge of adding an element to the `PopupStack`.
+
+v5 introduced focus management behaviors like `useInitialFocus` and `useReturnFocus` that was
+originally hard-coded into the `Modal`, but now `Modal` uses the hooks from the `popup` module. We
+noticed some application code that do custom focus management. Most of the special focus management
+code could be removed when using v5 the `Modal`.
```

Comment:
Is this supposed to say `v4 introduced focus management behaviors...`?

I'm going to suggest some alternate wording for this paragraph, but I won't use the GH suggestion feature in case I'm getting any details wrong - you can use however much of this makes sense to you.

```
v4 introduced focus management behaviors such as `useInitialFocus` and `useReturnFocus` that
were hard-coded into the `Modal`. In v5, `Modal` uses hooks from the `@workday/canvas-kit-react/popup`
module instead. These behaviors are no longer hard-coded and can be customized by passing
your own `PopupModel` to the `Modal` container component (see the [handleClose example](#handleclose)
below).

During our investigation of how `Modal` was currently being used, we noticed several instances
where custom focus management was necessary. Most of this special focus management should
no longer be necessary after migrating to v5.
```

Reply:
`v5` introduced the hooks available for export. The functionality was internal to Modal before that. In `v5`, the functionality was generalized to work for any popup.

Reply:
I'm planning to open another PR with doc updates. I wanted the PR mergable just in case while I worked more on docs.